media-sound/liblc3/liblc3-1.1.1-r1.ebuild

```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} pypy3 )
+inherit python-any-r1 toolchain-funcs meson-multilib
+
+DESCRIPTION="LC3 is an efficient low latency audio codec"
+HOMEPAGE="https://github.com/google/liblc3"
+SRC_URI="https://github.com/google/${PN}/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86"
+IUSE="test tools"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	test? (
+		$(python_gen_any_dep '
+			dev-python/numpy[${PYTHON_USEDEP}]
+			dev-python/scipy[${PYTHON_USEDEP}]
+		')
+	)
+"
+
+python_check_deps() {
+	python_has_version "dev-python/numpy[${PYTHON_USEDEP}]" &&
+	python_has_version "dev-python/scipy[${PYTHON_USEDEP}]"
+}
+
+pkg_setup() {
+	use test && python-any-r1_pkg_setup
+}
+
+multilib_src_prepare() {
+	use arm || rm -rf "test/arm" || die
+	use arm64 || rm -rf "test/neon" || die
+	default
+}
+
+multilib_src_configure() {
+	local emesonargs=(
+		-Dpython=false
+		$(meson_native_use_bool tools)
+	)
+	meson_src_configure
+}
+
+multilib_src_test() {
+	if multilib_is_native_abi; then
+		V= emake -C "${S}" test CC="$(tc-getCC)" \
+			CFLAGS:="${CPPFLAGS} ${CFLAGS} -I"$("${EPYTHON}" -c "import numpy;print(numpy.get_include())")""
+	else
+		ewarn "Skipping test for non-native ABI: ${ABI}"
+	fi
```

Comment:
I tried for non-native but failed:
```
x86_64-pc-linux-gnu-gcc -m32 -mfpmath=sse -fno-strict-overflow -Wsign-compare -DNDEBUG -march=native -Og -pipe -flto=auto -Werror=odr -Werror=strict-aliasing -Werror=lto-type-mismatch -g3 -ggdb3 -I/usr/lib/python3.12/site-packages/numpy/_core/include -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -I../src -I../include -I/usr/lib/python3.12/site-packages/numpy/_core/include -I/usr/include/python3.12 -c attdet_py.c -o build/temp.linux-x86_64-cpython-312/attdet_py.o -std=c11 -ffast-math
In file included from /usr/include/python3.12/Python.h:38,
                 from attdet_py.c:19:
/usr/include/python3.12/pyport.h:586:2: error: #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
  586 | #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
      |  ^~~~~
error: command '/usr/bin/x86_64-pc-linux-gnu-gcc' failed with exit code 1
```

Reply:
Can you test it "natively" in a 32-bit chroot as a one-off to make sure it works?

Reply:
I am not familiar with meson multilib but does this work?:
```
src_configure() {
	local emesonargs=(
		-Dpython=false
		$(meson_use tools)
	)
	meson-multilib_src_configure
}
```
Above error could also be a python-12 bug
In addition I think USE=tools must be false for 32 bits

Reply:
> I am not familiar with meson multilib but does this work?:
> 
> ```
> src_configure() {
> 	local emesonargs=(
> 		-Dpython=false
> 		$(meson_use tools)
> 	)
> 	meson-multilib_src_configure
> }
> ```
> 
> Above error could also be a python-12 bug In addition I think USE=tools must be false for 32 bits

tools is disabled for x86 (see meson-multilib.eclass docs) 

I think this is caused because python has no multilib support  




> Can you test it "natively" in a 32-bit chroot as a one-off to make sure it works?

@thesamesam Yes it works: https://paste.gentoo.zip/zkroffaK

Reply:
It's not a Python bug -- we have no 32-bit Python binary, headers, or library on 64-bit multilib systems.

Thanks, let's ship it.